pkg/xds/secrets/secrets.go

```diff
@@ -52,6 +52,9 @@ type Info struct {
 
 	OwnMesh        MeshInfo
 	OtherMeshInfos []MeshInfo
+	// this marks our info as having failed last time to get the mesh CAs that
+	// we wanted and so we should retry next time we want certs.
+	failedOtherMeshes bool
```

Comment:
nit: but how about we just use otherMeshInfos.
so this
```
			otherMeshInfos = append(otherMeshInfos, MeshInfo{
				MTLS: otherMesh.Spec.GetMtls(),
			})
```
is called after
```
			otherCa, _, err := s.caProvider.Get(context.Background(), otherMesh)
			if err != nil {
				log.Error(err, "could not get other mesh CA cert")
				continue
			}
```

Reply:
Yeah, I initially did it this way but it seems very implicit..